### PR TITLE
Test upgrades via github actions

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

If we use the snap process we can in theory test all the upgrades for a
given PR. This should highlight any potential issues during an upgrade
process.

## QA steps

Github test is valid

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/Ft1Igrs9/2954-github-upgrade-action-test

File: .github/workflows/upgrade.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          fi
+          sleep 10
+          attempt=$((attempt+1))
+          if [ "$attempt" -eq 48 ]; then
+              echo "Upgrade controller timed out"
+              exit 1
+          fi
+        done
+
+        PANIC=$(juju debug-log --replay --no-tail -m controller | grep "panic" || true)
```
Comment (hmlanigan) [inline .github/workflows/upgrade.yml:69]: 👍

**Review by hmlanigan — APPROVED** [review]
LGTM!

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**